src/app/reporting/ReportSchedulerImpl.cpp

```diff
@@ -59,23 +59,33 @@ void ReportSchedulerImpl::OnReadHandlerCreated(ReadHandler * aReadHandler)
     ScheduleReport(newTimeout, newNode);
 }
 
-/// @brief When a ReadHandler becomes reportable, schedule, verifies if the min interval of a handleris elapsed. If not,
-/// reschedule the report to happen when the min interval is elapsed. If it is, schedule an engine run.
-void ReportSchedulerImpl::OnBecameReportable(ReadHandler * aReadHandler)
+/// @brief When a ReadHandler's interval get changed after creation, update the min and max timestamps and reschedule the
+/// next timer callback.
+void ReportSchedulerImpl::OnReportingIntervalsChanged(ReadHandler * aReadHandler)
```

Comment:
i'm not sure if this is spec compliant and possible with the current implementation... Clients won't be updated on the new intervals...?

Reply:
+1 You cannot change min/max intervals without restarting a sub